eth/handler.go

```diff
@@ -286,7 +286,7 @@ func (pm *ProtocolManager) handle(p *peer) error {
 		}
 		if !fork.RequiredHash.IsEmpty() {
 			// Request the peer's fork block header for extra-dat
-			if err := p.RequestHeadersByNumber(fork.Block.Uint64(), 1, 0, false); err != nil {
+			if err := p.RequestHeadersByNumber(fork.Block.Uint64(), 1, 0, false); err != nil && err != core.ErrHashEmpty {
```

Comment:
Also, why is it possible that RequestHeadersByNumber would return `ErrHashEmpty` error... I think `RequestHeadersByNumber` just calls `p2p.Send` and the later just deliver the RequestHeaders message to the other peer and does not deal with the response.